config/hotreloadable.go

```diff
@@ -1,14 +1,64 @@
 package config
 
-import "time"
+import (
+	"strings"
+	"sync"
+	"time"
+)
 
 // RegisterIntConfigVariable registers int config variable
-func RegisterIntConfigVariable(defaultValue int, ptr *int, isHotReloadable bool, valueScale int, keys ...string) {
-	Default.RegisterIntConfigVariable(defaultValue, ptr, isHotReloadable, valueScale, keys...)
+// WARNING: a different order of keys will result in a different variable
+// Deprecated: use RegisterIntVar or RegisterReloadableIntVar instead
+func RegisterIntConfigVariable(defaultValue int, ptr *int, isHotReloadable bool, valueScale int, orderedKeys ...string) {
+	Default.RegisterIntConfigVariable(defaultValue, ptr, isHotReloadable, valueScale, orderedKeys...)
+}
+
+// RegisterIntVar registers a not hot-reloadable int config variable
+// WARNING: a different order of keys will result in a different variable
+func RegisterIntVar(defaultValue, valueScale int, orderedKeys ...string) *int {
```

Comment:
**Question:** wouldn't it be better to return values instead of pointers? Copies of primitives are simple, safe, straightforward & don't require heap.